### PR TITLE
clusterversion: create 25.1 version

### PR DESCRIPTION
**Pull request by RaduBerinde**
#### clusterversion: create 25.1 version

This change bumps the version on master to 25.1 without advancing the
previous release to 24.3 (the latter can only be done once we have a
published RC, or upgrade tests will fail). To allow this decoupling, a
few minor tweaks were needed (to `SupportedPreviousReleases` and to
the tool that generates the releases file.

Also made some minor improvements to the runbook and
sql-bootstrap-data. I will do a more thorough change to the runbook
that separates this step.

- [x] Add version key constant for new release, equal to `Latest`
- [x] Add start version
- [x] Update `pkg/build/version.txt` to the new version
- [x] Update the `scplan` rules in `pkg/sql/schemachanger/scplan/internal/rules` (already done)
- [x] Update releases file

Epic: REL-1322

### CODE REVIEW

File: pkg/cmd/release/update_releases.go [hunk 1]
```diff
@@ -334,12 +334,13 @@ func generateRepositoriesFile(versions ...string) error {
...
 		for cfgKey, cfg := range cfgKeys {
 			url := fmt.Sprintf("https://binaries.cockroachdb.com/cockroach-v%s.%s.tgz", v, cfg)
+			fmt.Printf("getting %s\n", url)
 			resp, err := client.Get(context.Background(), url)
 			if err != nil {
 				return fmt.Errorf("could not download cockroach release: %w", err)
 			}
 			if resp.StatusCode != http.StatusOK {
-				return fmt.Errorf("unexpected status code %d when downloading %s", resp.StatusCode, url)
+				return fmt.Errorf("unexpected status code %d when downloading %s; consider turning off NetSkope", resp.StatusCode, url)
```
Comment (dt) [inline pkg/cmd/release/update_releases.go:343]: I believe we avoid mentioning these sorts of specifics. Maybe say "disabling any web traffic monitoring tools"

**Review by dt — APPROVED** [review]

File: pkg/cmd/release/update_releases.go (hunk 1 above)
Comment (RaduBerinde) [reply pkg/cmd/release/update_releases.go]: Undid this change, it looks like it was a transient issue on my machine

**Comment by RaduBerinde** [comment]
CC @spilchen for the second commit.

File: pkg/clusterversion/README.md [hunk 2]
```diff
@@ -183,7 +183,10 @@ avoid large changes on `master` which might cause merge conflicts for backports.
 
 - [ ] Update `PreviousRelease` constant
```
Comment (celiala) [inline pkg/clusterversion/README.md:185]: thank you for keeping the runbook up-to-date!

File: pkg/cmd/release/update_releases.go [hunk 3]
```diff
@@ -126,10 +126,10 @@ func processReleaseData(data []Release) map[string]release.Series {
...
 		}
 
-		// For the purposes of the cockroach_releases file, we are only
-		// interested in beta and rc pre-releases, as we do not support
-		// upgrades from alpha releases.
-		if pre := v.PreRelease(); pre != "" && !strings.HasPrefix(pre, "rc") && !strings.HasPrefix(pre, "beta") {
+		// For the purposes of the cockroach_releases file, we are only interested
+		// in rc pre-releases, as we do not support upgrades from alpha or beta
+		// releases.
+		if pre := v.PreRelease(); pre != "" && !strings.HasPrefix(pre, "rc") {
```
Comment (celiala) [inline pkg/cmd/release/update_releases.go:132]: nice fix/solution 👍🏼

File: pkg/cmd/release/update_releases_test.go [hunk 4]
```diff
@@ -39,10 +39,6 @@ func Test_processReleaseData(t *testing.T) {
 			Predecessor: "22.2",
 			Withdrawn:   []string{"23.1.0"},
 		},
-		"23.2": {
```
Comment (celiala) [inline pkg/cmd/release/update_releases_test.go:42]: to confirm, this removal/cleanup is more related to advancing `MinSupported` to 24.3 (than to creating 24.3 and 25.1 start version gates, right?)

(mostly to clarify for when we refer to this for the next release cycle/task)

File: pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion_test.go [hunk 5]
```diff
@@ -244,7 +245,7 @@ func Test_choosePreviousReleases(t *testing.T) {
 			arch:              vm.ArchAMD64,
 			numUpgrades:       3,
 			enableSkipVersion: true,
-			expectedReleases:  []string{"23.2.4", "24.1.1", "24.2.2"},
+			expectedReleases:  []string{"23.1.17", "23.2.4", "24.1.1"},
```
Comment (celiala) [inline pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion_test.go:248]: x-referencing [note from MinSupport bump comment](https://github.com/cockroachdb/cockroach/pull/134340#pullrequestreview-2419388116) (for when we do this task for the next cycle), that this change is a result of the randomized sequence of versions that the test decides on with that seed.

File: pkg/testutils/release/cockroach_releases.yaml [hunk 6]
```diff
@@ -29,6 +29,5 @@
   withdrawn:
   - 24.2.1
   predecessor: "24.1"
-"24.3":
```
Comment (celiala) [inline pkg/testutils/release/cockroach_releases.yaml:32]: we should keep 24.3, yes?

**Review by celiala — COMMENTED** [review]
I had one question around keeping 24.3 (instead of replacing) for `pkg/testutils/release/cockroach_releases.yaml`. 

everything else made sense to me.

File: pkg/cmd/release/update_releases_test.go (hunk 4 above)
Comment (RaduBerinde) [reply pkg/cmd/release/update_releases_test.go:42]: This is just a test for the updated releases code, which we updated to ignore beta builds. This test runs against fixed data, the versions should be considered fictitious.

File: pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion_test.go (hunk 5 above)
Comment (RaduBerinde) [reply pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion_test.go:248]: Right, and it's all based on the testPredecessorMapping (not the actual release data)

**Review by celiala — APPROVED** [review]
LGTM - thank you!

**Comment by RaduBerinde** [comment]
TFTRs!

bors r+

**Pull request merged**